<launch>
    <arg name="gazebo_sim" default="false"/>
    <arg name="global_frame_id" default="world" unless="$(arg gazebo_sim)"/>
    <arg name="global_frame_id" default="map" if="$(arg gazebo_sim)"/>
    <arg name="drone_frame_id" default="matrice" unless="$(arg gazebo_sim)"/>
    <arg name="drone_frame_id" default="firefly/base_link" if="$(arg gazebo_sim)"/>
    <node pkg="matrice_traj_tracker" type="matrice_marsupial_traj_tracker_node" name="matrice_marsupial_traj_tracker_node" output="screen">
    <!-- node parameters -->
        <param name="takeoff_height" type="double" value="3.0"/>
        <param name="max_vx" type="double" value="0.5"/>
        <param name="max_vy" type="double" value="0.5"/>
        <param name="max_vz" type="double" value="0.4"/>
        <param name="max_ry" type="double" value="0.5"/>
        <param name="min_vx" type="double" value="0.2"/>
        <param name="min_vy" type="double" value="0.2"/>
        <param name="min_vz" type="double" value="0.2"/>
        <param name="min_ry" type="double" value="0.2"/>
        <param name="watchdog_freq" type="double" value="7.0"/>
        <param name="testing" value="false" />
        <param name="gazebo_sim" value="$(arg gazebo_sim)" />
        <param name="drone_model" value="m210" />
        <param name="drone_frame" value="$(arg drone_frame_id)" />
        <param name="global_frame_id" value="$(arg global_frame_id)"/>
        <param name="_frame" value="$(arg global_frame_id)" />
        <param name="speed_reference_mode" value="true" />
        <remap from="/dji_sdk/rc" to="/firefly/joy" />
        <remap from="/dji_sdk/gps_position" to="/firefly/gps" />
    </node>

</launch>

